### PR TITLE
Feature/multiloader

### LINKED ISSUE #23: When running a LAN server, users cannot join unless they have Lighty installed
As the title says users who try to join the server cannot unless they also have the mod installed. This extends to lan servers proxied through e4mc although I did try disabling that mod just in case. 

Fabric 0.14.21 | Minecraft 1.20 
Lighty 2.00 for MC 1.20 
Prism Launcher Flatpak on Fedora Linux

### PR DESCRIPTION
**Pull request by justliliandev**
uhm yeah, multiloader stuff I guess

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -1,80 +1,48 @@
...
+allprojects {
+    apply plugin: "java"
+    apply plugin: "architectury-plugin"
+    apply plugin: "maven-publish"
 
-jar {
-    from("LICENSE") {
-        rename { "${it}_${project.archivesBaseName}" }
+    archivesBaseName = rootProject.archives_base_name
+    version = rootProject.mod_version
```
Comment (andi-makes) [inline build.gradle]: For the old version string I had "${project.mod_version}+${project.minecraft_version}", any reason this should be changed? Does this gradle handle putting the minecraft version at the end anywhere else?

File: common/src/main/java/dev/schmarrn/lighty/UtilDefinition.java [hunk 2]
```diff
@@ -0,0 +1,17 @@
+package dev.schmarrn.lighty;
+
+import net.minecraft.client.KeyMapping;
+import java.nio.file.Path;
+import java.util.ServiceLoader;
+
+public interface UtilDefinition {
+    UtilDefinition INSTANCE = load();
+    KeyMapping registerKeyBinding(KeyMapping mapping);
+
```
Comment (andi-makes) [inline common/src/main/java/dev/schmarrn/lighty/UtilDefinition.java:13]: one newline too much

File: common/src/main/java/dev/schmarrn/lighty/UtilDefinition.java [hunk 3]
```diff
@@ -0,0 +1,17 @@
+package dev.schmarrn.lighty;
+
+import net.minecraft.client.KeyMapping;
+import java.nio.file.Path;
+import java.util.ServiceLoader;
+
+public interface UtilDefinition {
```
Comment (andi-makes) [inline common/src/main/java/dev/schmarrn/lighty/UtilDefinition.java:10]: Maybe add some comments explaining *why* we need to have this Interface, how it is supposed to be used

It seems like it is something for Keybinds, maybe some kind of abstraction for forge and fabric? Just putting a comment linking to existing documentation is also fine by me

File: common/src/main/java/dev/schmarrn/lighty/Lighty.java [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+        LOGGER.info("Let there be {}", MOD_NAME);
+
+        Config.init();
+        CarpetMode.init();
+        NumberMode.init();
+        BoringCrossMode.init();
+        KeyBind.init();
+    }
+
+    public static void postLoad() {
```
Comment (andi-makes) [inline common/src/main/java/dev/schmarrn/lighty/Lighty.java:32]: Add comment something like "gets called after all Lighty Modes are loaded"

File: common/src/main/java/dev/schmarrn/lighty/UtilDefinition.java (hunk 3 above)
Comment (andi-makes) [reply common/src/main/java/dev/schmarrn/lighty/UtilDefinition.java:10]: Or putting some javadoc above the methods explaining (roughly, not in detail) what they do on fabric/forge, if they result in the same behavior and just are implemented differently

File: common/src/main/java/dev/schmarrn/lighty/event/KeyBind.java [hunk 5]
```diff
@@ -44,9 +43,7 @@ public static void handleKeyBind(Minecraft client) {
         KeyBind.oldToggleState = newToggleState;
     }
 
-    public static void init() {
-        ClientTickEvents.END_CLIENT_TICK.register(KeyBind::handleKeyBind);
-    }
+    public static void init() {}
```
Comment (andi-makes) [inline common/src/main/java/dev/schmarrn/lighty/event/KeyBind.java]: Putting a comment something like "does nothing besides making sure the keybinds get registered using UtilDefinition"

File: common/src/main/java/dev/schmarrn/lighty/mode/CarpetMode.java [hunk 6]
```diff
@@ -91,42 +82,58 @@ public void compute(ClientLevel world, BlockPos pos, BufferBuilder builder) {
...
+        double x = pos.getX();
+        double y = pos.getY() + 1 + offset;
+        double z = pos.getZ();
+        //TOP
+        builder.vertex(x, y + 1 / 16f, z).uv(0, 0).color(color).endVertex();
+        builder.vertex(x, y + 1 / 16f, z + 1).uv(0, 1).color(color).endVertex();
+        builder.vertex(x + 1, y + 1 / 16f, z + 1).uv(1, 1).color(color).endVertex();
+        builder.vertex(x + 1, y + 1 / 16f, z).uv(1, 0).color(color).endVertex();
+        if (offset > 0.001f)
+            pos = pos.above();
```
Comment (andi-makes) [inline common/src/main/java/dev/schmarrn/lighty/mode/CarpetMode.java:95]: please put {} and a comment something like "needed for face culling functionality below"

File: forge/src/main/resources/META-INF/mods.toml [hunk 7]
```diff
@@ -0,0 +1,28 @@
+modLoader = "javafml"
+loaderVersion = "[46,)"
+issueTrackerURL = "https://github.com/SchmarrnDevs/Lighty/issues"
+license = "Apache-2.0"
+
+[[mods]]
+modId = "lighty"
+version = "${version}"
+displayName = "Lighty"
+authors = "andi_makes"
```
Comment (andi-makes) [inline forge/src/main/resources/META-INF/mods.toml]: come on, add yourself there :P 
Also feel free to do so in the fabric.mod.json

File: forge/src/main/java/dev/schhmarrn/lighty/forge/LightyForge.java [hunk 8]
```diff
@@ -0,0 +1,55 @@
...
+import org.slf4j.LoggerFactory;
+
+@Mod(Lighty.MOD_ID)
+public class LightyForge {
+
+    public LightyForge() {
+        Lighty.init();
+    }
+
+    @Mod.EventBusSubscriber(bus = Mod.EventBusSubscriber.Bus.MOD, value = Dist.CLIENT)
```
Comment (andi-makes) [inline forge/src/main/java/dev/schhmarrn/lighty/forge/LightyForge.java:28]: Please put a comment with a link to documentation where someone not experienced in forge things (me) can look at the forge magic (applies to all other forge magic thingies as well)

File: forge/src/main/java/dev/schhmarrn/lighty/forge/UtilForgeImpl.java [hunk 9]
```diff
@@ -0,0 +1,35 @@
...
+import net.minecraftforge.client.event.RegisterKeyMappingsEvent;
+import net.minecraftforge.eventbus.api.SubscribeEvent;
+import net.minecraftforge.fml.common.Mod;
+import net.minecraftforge.fml.loading.FMLPaths;
+
+import java.nio.file.Path;
+import java.util.ArrayList;
+import java.util.List;
+
+@Mod.EventBusSubscriber(value = Dist.CLIENT, bus = Mod.EventBusSubscriber.Bus.MOD)
```
Comment (andi-makes) [inline forge/src/main/java/dev/schhmarrn/lighty/forge/UtilForgeImpl.java:15]: Magic forge thingie

File: forge/src/main/resources/pack.mcmeta [hunk 10]
```diff
@@ -0,0 +1,6 @@
+{
+  "pack": {
+    "description": "Example Mod",
```
Comment (andi-makes) [inline forge/src/main/resources/pack.mcmeta]: this seems old? This isn't an example mod for sure I think

File: gradle.properties [hunk 11]
```diff
@@ -1,15 +1,16 @@
...
 mod_version=2.0.0
 maven_group=dev.schmarrn
 archives_base_name=lighty
+
+fabric_loader_version=0.14.21
+fabric_api_version=0.83.1+1.20.1
+
+forge_version=1.20.1-47.0.1
+
+
```
Comment (andi-makes) [inline gradle.properties]: remove all but one newline pls

**Review by andi-makes — CHANGES_REQUESTED** [review]
Thank you very much for this great PR! Great work!

There are just some minor nitpicks, mostly regarding formatting and adding some comments explaining what some things do and/or linking to existing documentation.

Regardless, your changes were really clear and I was able to understand most of the things, just the forge magic thingies remain a mystery to me.

I guess I'll need to update ci/cd, but I'll look into that myself (unless you have an existing solution somewhere)

Great Work! Again, thank you very much!

**Comment by andi-makes** [comment]
Also, this PR theoretically fixes #23

**Comment by andi-makes** [comment]
~~But I have no Idea how to link this to the issue so it gets closed automatically :/~~

Nevermind, figured it out

File: forge/src/main/java/dev/schhmarrn/lighty/forge/LightyForge.java (hunk 8 above)
Comment (justliliandev) [reply forge/src/main/java/dev/schhmarrn/lighty/forge/LightyForge.java:28]: The documentation is already present in the javadoc of the anotation

Comment (andi-makes) [reply forge/src/main/java/dev/schhmarrn/lighty/forge/LightyForge.java:28]: But for that I'd need to open up the IDE, which I didn't for now

**Review by andi-makes — COMMENTED** [review]
lgtm, thank you! I'll try building it locally now, if everything works I'll merge

**Review by andi-makes — APPROVED** [review]
works great!

**Pull request merged**